content/en/blog/_posts/2023-04-11-topology-spread-features.md

```diff
@@ -0,0 +1,140 @@
+---
+layout: blog
+title: "Kubernetes 1.27: More fine-grained pod topology spread policies reached beta"
+date: 2023-04-11
+slug: fine-grained-pod-topology-spread-features-beta
+evergreen: true
+---
+
+**Authors:** [Alex Wang](https://github.com/denkensk)(Shopee), [Kante Yin](https://github.com/kerthcet)(DaoCloud), [Kensei Nakada](https://github.com/sanposhiho)(Mercari)
+
+In Kubernetes v1.19, [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) went to GA.
+It is the feature to control how Pods are spread to each failure-domain (regions, zones, nodes etc).
+
+As time passes, we've got further feedbacks from users,
+and we're actively working on improving the Topology Spread via three KEPs from v1.25.
+All of these features have reached beta in Kubernetes v1.27 and been enabled by default.
+
+This blog post is going to introduce each feature and the usecase/issue behind them.
+
+## KEP-3022: min domains in Pod Topology Spread
+
+Pod Topology Spread has the `maxSkew` parameter to define the degree to which Pods may be unevenly distributed. 
+
+But, there wasn't a way to control the number of domains over which we should spread. 
+Some users want to force spreading Pods over a minimum number of domains, and if there aren't enough already present, make the cluster-autoscaler provision them.
+
+Then, we introduced the `minDomains` parameter in the Pod Topology Spread. 
+Via `minDomains` parameter, you can define the minimum number of domains. 
+
+For example, there are 3 Nodes with the enough capacity, 
+and newly created replicaset has the following `topologySpreadConstraints` in template.
+
+```yaml
+topologySpreadConstraints:
+- maxSkew: 1
+  minDomains: 5 # requires 5 Nodes at least.
+  whenUnsatisfiable: DoNotSchedule # minDomains is valid only when DoNotSchedule is used.
+  topologyKey: kubernetes.io/hostname 
+  labelSelector:
+    matchLabels:
+        foo: bar
+```
+
+This case, 3 Pods will be scheduled to those 3 Nodes,
+but other 2 Pods from this replicaset will be unschedulable until more Nodes join the cluster.
+
+The cluster autoscaler provisions new Nodes based on these unschedulable Pods,
+and as a result, the replicas are finally spread over 5 Nodes.
+
+## KEP-3094: Take taints/tolerations into consideration when calculating PodTopologySpread skew
+
+Before this, when we deploy a pod with `podTopologySpread` configured, we'll take all
+affinity nodes(satisfied with pod nodeAffinity and nodeSelector) into consideration
+in filtering and scoring, but a node with pod untolerated taint may also be a candidate
```

Comment:
```suggestion
in filtering and scoring. However, a Node which the incoming Pod doesn't tolerate taint may also be a candidate
```

Reply:
Although I didn't quite understand the example

Reply:
Polished the description.